### PR TITLE
transmission: fix transmission-web dependencies

### LINKED ISSUE #2835: selecting transmission-daemon-openssl does not allow transmission-web to be selected
AFAIK the -polarssl variant is the same.

### PR DESCRIPTION
**Pull request by mstorchak**
Fixes #2835

Signed-off-by: Maxim Storchak <m.storchak@gmail.com>

### CODE REVIEW

**Comment by mstorchak** [comment]
@obsy

**Comment by obsy** [comment]
Acked-by: Cezary Jackiewicz <cezary@eko.one.pl>

File: net/transmission/Makefile [hunk 1]
```diff
@@ -94,7 +93,7 @@ endef
 define Package/transmission-web
   $(call Package/transmission/template)
   TITLE+= (webinterface)
-  DEPENDS:=transmission-daemon
+  DEPENDS:=transmission-daemon-openssl||PACKAGE_transmission-daemon-polarssl
```
Comment (champtar) [inline net/transmission/Makefile]: I think "PACKAGE_" is not needed here

Comment (champtar) [inline net/transmission/Makefile]: just read the issue, maybe put "PACKAGE_" on both for more consistency?

Comment (nbd168) [inline net/transmission/Makefile]: The dependency should be transmission-daemon. The -openssl and -polarssl packages should use PROVIDES:=transmission-daemon

Comment (champtar) [inline net/transmission/Makefile]: @nbd168 this seems to not work, can you take a expert look

**Comment by mstorchak** [comment]
When grepping through makefiles I found this example of alternative dependencies:
`./net/pppossh/Makefile:  DEPENDS:=+ppp +resolveip @(PACKAGE_dropbear||PACKAGE_openssh-client)`.
Would it be better to rewrite it like this?

File: net/transmission/Makefile (hunk 1 above)
Comment (mstorchak) [inline net/transmission/Makefile]: When grepping through makefiles I found this example of alternative dependencies:
`./net/pppossh/Makefile: DEPENDS:=+ppp +resolveip @(PACKAGE_dropbear||PACKAGE_openssh-client).`
Would it be better to rewrite it like this?

Comment (champtar) [inline net/transmission/Makefile]: @mstorchak ```DEPENDS:=PACKAGE_transmission-daemon-openssl||PACKAGE_transmission-daemon-polarssl``` if it work is enough for me, but PROVIDES should work

Comment (mstorchak) [inline net/transmission/Makefile]: `DEPENDS:=PACKAGE_transmission-daemon-openssl||PACKAGE_transmission-daemon-polarssl` doesn't work, only -polarssl opens -web.
The same result for PROVIDES/DEPENDS, which looks like a fundamental problem of the configuration system.
The only two variants that work are `DEPENDS:=transmission-daemon-openssl||PACKAGE_transmission-daemon-polarssl` and a more symmetrical one `DEPENDS:=@(PACKAGE_transmission-daemon-openssl||PACKAGE_transmission-daemon-polarssl)`

**Comment by mstorchak** [comment]
I've updated the PR to make dependency symmetrical.
@obsy, @champtar, @nbd168, please review.

**Comment by champtar** [comment]
I'll try to understand why provides doesn't work later, in the mean time merging this
Thanks @mstorchak

**Comment by dfc-d** [comment]
On Mon, 2016-06-13 at 04:54 -0700, champtar wrote:
> I'll try to understand why provides doesn't work later, in the mean
> time merging this
> Thanks @mstorchak 

With PROVIDES, opkg installs the last matching provider, unless a
provider is *already* installed.  For IB this can be arranged by making
sure you supply a provider first in a PACKAGES="package1 package2 ..."
makefile variable.  Probably in the case of a non ImageBuilder build
there is no way to ensure the necessary order of packages being
installed into the image, leading to this provides installing both the
the one you selected and the default (last one opkg finds (the last bit
is what I discovered from reading the opkg code)).

Regards,

Daniel

**Comment by dfc-d** [comment]
https://dev.openwrt.org/ticket/21344 is a ticket I opened on OpenWrt for this issue.  It has been proposed that and updated version of opkg would solve the issue, however Jow points out here: https://lists.openwrt.org/pipermail/openwrt-devel/2016-April/040899.html that a great deal of work is required to make it useful to OpenWrt/LEDE because of new upstream (Yocto) making it too large.

**Pull request merged**